src/client/app/containers/app.ts

```diff
@@ -190,22 +190,24 @@ export class AppComponent {
                     this._store.dispatch(new GitHub.UpdateGistAction(this.snippet));
                 }
                 else if (isPublic) {
-                    this._effects.alert(this.strings.sharePublicSnippetConfirm, `${this.strings.share} ${this.snippet.name}`, this.strings.share, this.strings.cancelButtonLabel)
-                        .then((result: string) => {
-                            if (result !== this.strings.cancelButtonLabel) {
-                                this._store.dispatch(new GitHub.SharePublicGistAction(this.snippet));
-                            }
+                    if (this.snippet.gist) {
+                        let result = await this._effects.alert(this.strings.sharePublicSnippetConfirm, `${this.strings.share} ${this.snippet.name}`, this.strings.share, this.strings.cancelButtonLabel);
+                        if (result === this.strings.cancelButtonLabel) {
```

Comment:
Isn't this "`if (result === this.strings.cancelButtonLabel) {`" statement shared between both public and private (and there is nothing else other than those two?)?  Should you move it out to outside the if statement?